pkg/chaosdaemon/tc_server.go

```diff
@@ -74,33 +75,37 @@ func generateQdiscArgs(action string, qdisc *pb.Qdisc) ([]string, error) {
 }
 
 func getAllInterfaces(ctx context.Context, log logr.Logger, pid uint32, enterNS bool) ([]string, error) {
-	processBuilder := bpm.DefaultProcessBuilder("ip", "-j", "addr", "show").SetContext(ctx)
-	if enterNS {
-		processBuilder = processBuilder.SetNS(pid, bpm.NetNS)
-	}
-	cmd := processBuilder.Build(ctx)
-	ipOutput, err := cmd.CombinedOutput()
-	if err != nil {
-		return []string{}, err
-	}
-	var data []map[string]interface{}
-
-	err = json.Unmarshal(ipOutput, &data)
-	if err != nil {
-		return []string{}, err
-	}
-
 	var ifaces []string
-	for _, iface := range data {
-		name, ok := iface["ifname"]
-		if !ok {
-			return []string{}, errors.New("fail to read ifname from ip -j addr show")
+	if enterNS {
+		ipOutput, err := bpm.DefaultProcessBuilder("ip", "-j", "addr", "show").SetNS(pid, bpm.NetNS).SetContext(ctx).Build(ctx).CombinedOutput()
+		if err != nil {
+			return []string{}, err
 		}
+		var data []map[string]interface{}
 
-		ifaces = append(ifaces, name.(string))
+		err = json.Unmarshal(ipOutput, &data)
+		if err != nil {
+			return []string{}, err
+		}
+		for _, iface := range data {
+			name, ok := iface["ifname"]
+			if !ok {
+				return []string{}, errors.New("fail to read ifname from ip -j addr show")
+			}
+			ifaces = append(ifaces, name.(string))
+		}
+		log.Info("get interfaces from ip command", "ifaces", ifaces)
+	} else {
+		interfaces, err := net.Interfaces()
+		if err != nil {
+			return []string{}, errors.New("fail to read ifname from net.Interfaces()")
+		}
+		for _, iface := range interfaces {
+			ifaces = append(ifaces, iface.Name)
+		}
+		log.Info("get interfaces from net.Interfaces()", "ifaces", ifaces)
```

Comment:
Move this log out of the `if` condition.

Reply:
There are two logs in branches, and they are different in sources ([L97](https://github.com/FingerLeader/chaos-mesh/blob/0212a5f7879da2a0baa9eeccc24cb33df0a868d5/pkg/chaosdaemon/tc_server.go#L97))

Reply:
Sorry. I understand :facepalm: